test/wasi/hello_world.wast

```diff
@@ -1,24 +1,23 @@
 (module
-  (import "wasi_snapshot_preview1" "fd_write" (func $__wasi_fd_write (param i32 i32 i32 i32) (result i32))) 
+  (import "wasi_snapshot_preview1" "fd_write" (func $wasi_fd_write (param i32 i32 i32 i32) (result i32))) 
   (memory 1)
   
   (export "memory" (memory 0))
-  (export "_start" (func $_start))
+  (export "hello_world" (func $hello_world))
   (data (i32.const 0) "Hello World!\n")
 
-  (func $_start (result i32)
-    (i32.store (i32.const 24) (i32.const 13)) ;; Lenght of "Hello World!\n"
-    (i32.store (i32.const 20) (i32.const 0))  ;; memory offset of "Hello World!\n"
-    (call $__wasi_fd_write
+  (func $hello_world
+    (i32.store (i32.const 100) (i32.const 0))
+    (i32.store (i32.const 112) (i32.const 13))
+    (i32.store (i32.const 200) (i32.const 0))
+    (call $wasi_fd_write
           (i32.const 1)  ;;file descriptor
-          (i32.const 20) ;;offset of str offset
-          (i32.const 1)  ;;iovec length
-          (i32.const 30) ;;result offset
+          (i32.const 100) ;;offset of str offset
+          (i32.const 2)  ;;iovec length
```

Comment:
2 or 112? The whole parameter passing looks confusing. A pointer to a pointer is passed? Looks unnecessary at first.

Reply:
It was a mistake on my part, I left those in when debugging. A pointer to a pointer is passed because the pointer of a string is followed by the lenght of that string. (ciovec construction in wasi).

It should be 1. And the offset above should be 104 instead of 112.

It should be fixed now.